src/Chapter3.hs

```diff
@@ -909,7 +980,16 @@ Implement instances of "Append" for the following types:
 -}
 class Append a where
     append :: a -> a -> a
-
+newtype Gold = Gold Int
+instance Append Gold where
+  append (Gold i) (Gold j) = Gold (i+j)
+instance Append [a] where
+  append i j = i++j
+instance (Append a) => Append (Maybe a) where
+  append (Just i) (Just j) = Just (append i j)
+  append (Just i) Nothing = Just i
+  append Nothing (Just j) = Just j
+  append _ _ = Nothing       
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂